modules/cli/cli.go

```diff
@@ -148,15 +144,6 @@ func (c *CLI) Metadata() metadata.Info {
 // Must be called after all additional flags in the FlagSet() are defined and before flags are accessed
 // by the program. It returns config for controller manager.
 func (c *CLI) Parse(arguments []string) manager.Config {
-	developmentModeEnabled := manager.DefaultConfig().DevelopmentMode
-	// TODO: clean env handling https://github.com/Kong/gateway-operator-archive/issues/19
-	if os.Getenv(envVarFlagPrefix+"DEVELOPMENT_MODE") == "true" ||
-		os.Getenv("CONTROLLER_DEVELOPMENT_MODE") == "true" {
-		developmentModeEnabled = true
-	}
```

Comment:
I think we should warn our users who still use either of these envs so that they migrate to using the new (introduced in this PR).

How about printing a log message stating that this env is now ineffective and users should use the new envs?

Reply:
PTAL: https://github.com/Kong/gateway-operator/pull/1435/commits/70046304700aae6bb81b669a2d87c147e0faa080